zerver/lib/actions.py

```diff
@@ -8008,7 +8008,21 @@ def check_add_realm_emoji(
     realm_emoji.file_name = emoji_file_name
     realm_emoji.is_animated = is_animated
     realm_emoji.save(update_fields=["file_name", "is_animated"])
-    notify_realm_emoji(realm_emoji.realm, realm.get_emoji())
+
+    realm_emoji_dict = realm.get_emoji()
+    RealmAuditLog.objects.create(
+        realm=realm,
+        acting_user=author,
+        event_type=RealmAuditLog.REALM_EMOJI_ADDED,
+        event_time=timezone_now(),
+        extra_data=orjson.dumps(
+            {
+                "realm_emoji": dict(sorted(realm_emoji_dict.items())),
```

Comment:
Have sorted this because the tests were failing as there was not a specific order in which `realm.get_emoji()` returns the emojis. We can also modify the tests in some way, but this seemed better to keep tests robust.